vectordraw/vectordraw.py

```diff
@@ -0,0 +1,312 @@
+            'vectors': self.vectors_json,
+            'points': self.points_json,
+            'expected_result': self.expected_result_json
+        }
+
+    @property
+    def user_state(self):
+        user_state = self.answer
+        if self.result:
+            user_state['result'] = self.result
+        return user_state
+
+    @property
+    def background(self):
+        return {
+            'src': self.background_url,
+            'width': self.background_width,
+            'height': self.background_height,
+        }
+
+    @property
+    def vectors_json(self):
+        return json.loads(self.vectors)
+
+    @property
+    def points_json(self):
+        return json.loads(self.points)
+
+    @property
+    def expected_result_json(self):
+        return json.loads(self.expected_result)
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def student_view(self, context=None):
+        """
+        The primary view of the VectorDrawXBlock, shown to students
+        when viewing courses.
+        """
+        context['self'] = self
+        fragment = Fragment()
+        fragment.add_content(loader.render_template('static/html/vectordraw.html', context))
+        fragment.add_css_url("//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css")
+        fragment.add_css(self.resource_string('static/css/vectordraw.css'))
+        fragment.add_javascript_url("//cdnjs.cloudflare.com/ajax/libs/jsxgraph/0.98/jsxgraphcore.js")
+        fragment.add_javascript(self.resource_string("static/js/src/vectordraw.js"))
+        fragment.initialize_js('VectorDrawXBlock', {"settings": self.settings, "user_state": self.user_state})
+        return fragment
+
+    @XBlock.json_handler
+    def check_answer(self, data, suffix=''):
+        """
+        Check and persist student's answer to this vector drawing problem.
+        """
+        # Save answer
+        self.answer = dict(
+            vectors=data["vectors"],
+            points=data["points"]
+        )
+        # Compute result
+        grader = Grader()
+        result = grader.grade(data)
+        # Save result
+        self.result = result
+        # Publish grade data
+        score = 1 if result["ok"] else 0
+        self.runtime.publish(self, 'grade', dict(value=score, max_value=1))
+        return {
+            "result": result,
+        }
+
+    # TO-DO: change this to create the scenarios you'd like to see in the
+    # workbench while developing your XBlock.
+    @staticmethod
+    def workbench_scenarios():
+        """A canned scenario for display in the workbench."""
+        return [
+            ("VectorDrawXBlock",
+             """<vertical_demo>
+                <vectordraw/>
+                <vectordraw/>
+                <vectordraw/>
+                </vertical_demo>
+             """),
+        ]
```

Comment:
This scenarios doesn't make too much sense yet (though it would likely work, showing three copies of a vectordraw XBlock with all settings at their default values).